Seneca-CDOT/telescope
# Fixes #3208 install missing deps in src/docs

## Issue This PR Addresses



Related to #3208

## Type of Change




## Description



This PR does the following:
- moves `devDependencies` underneath `dependencies` for better legibility
- installs `@algolia/client-search`, `@types/react`, `typescript`, and `unist-util-visit` because they are listed as missing dependencies
- downgrades `mermaid` from v8.14 to v8.11 because v8.14 is not the version `mdx-mermaid` needs.

## Steps to test the PR



1. Add my repository as a remote: `git remote add cindyledev https://github.com/cindyledev/telescope.git`
2. Fetch my branches: `git fetch cindyledev`
3. Checkout my branch: `git checkout issue-3208-b`
4. Install dependencies using pnpm in root: `pnpm install`
5. You should not see:

```
src/docs
├─┬ @docusaurus/core
│ └─┬ react-dev-utils
│   └─┬ fork-ts-checker-webpack-plugin
│     └── ✕ missing peer typescript@">= 2.7"
├─┬ @docusaurus/preset-classic
│ ├─┬ @docusaurus/plugin-debug
│ │ └─┬ react-json-view
│ │   └─┬ react-textarea-autosize
│ │     └─┬ use-latest
│ │       ├── ✕ missing peer @types/react@"*"
│ │       └─┬ use-isomorphic-layout-effect
│ │         └── ✕ missing peer @types/react@"*"
│ └─┬ @docusaurus/theme-search-algolia
│   └─┬ @docsearch/react
│     ├── ✕ missing peer @types/react@">= 16.8.0 < 18.0.0"
│     └─┬ @algolia/autocomplete-preset-algolia
│       └── ✕ missing peer @algolia/client-search@^4.9.1
└─┬ mdx-mermaid
  ├── ✕ missing peer unist-util-visit@^2.0.0
  └── ✕ unmet peer mermaid@">= 8.11.0 < 8.12.0": found 8.14.0
Peer dependencies that should be installed:
  @algolia/client-search@^4.9.1    @types/react@">=16.8.0 <18.0.0"  typescript@">= 2.7"              unist-util-visit@^2.0.0          
```

Note: Depending on your OS, you may or may not even see these warnings. I get these warnings using AWS EC2 Amazon Linux 2 AMI. Also if you do not see the errors on the `master` branch, try deleting the `pnpm-lock.yaml` file.

## Checklist
